.github/workflows/ci-build.yml

```diff
@@ -6,49 +6,59 @@ on:
   pull_request:
   workflow_dispatch:
 
-env:
-  CARGO_TERM_COLOR: always
-
 jobs:
-  janus_server_build:
+  janus_build:
     runs-on: ubuntu-latest
+    env:
+      CARGO_INCREMENTAL: 0
+      CARGO_TERM_COLOR: always
+      RUSTFLAGS: "-D warnings"
     steps:
     - uses: actions/checkout@v3
-    - name: cache
-      uses: actions/cache@v3
-      with:
-        path: |
-          ~/.cargo/bin/
-          ~/.cargo/registry/index/
-          ~/.cargo/registry/cache/
-          ~/.cargo/git/db/
-          target/
-        key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+    - name: Setup Go toolchain
+      uses: actions/setup-go@v3
     - name: Install Kind
       # We should always make sure that the `kind` CLI we install is from the
       # same release as the node image version used by
       # `janus_core::test_util::kubernetes::EphemeralCluster`
       run: go install sigs.k8s.io/kind@v0.14.0
-    - name: lint
+    - name: Install Rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        profile: minimal
+        override: true
+    - name: Cache dependencies
+      uses: Swatinem/rust-cache@v2
+    - name: Build
+      run: cargo test --no-run --locked
+    - name: Test
+      run: cargo test
+  
+  janus_lints:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_INCREMENTAL: 0
+      CARGO_TERM_COLOR: always
+      RUSTDOCFLAGS: "-D warnings"
+      RUSTFLAGS: "-D warnings"
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install Rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        profile: minimal
+        override: true
+        components: clippy, rustfmt
+    - name: Format
       run: cargo fmt --message-format human -- --check
-    - name: clippy
-      run: cargo clippy --workspace --all-targets -- -D warnings
-    - name: check, all features
-      run: cargo check --all-features
-    - name: build
-      run: cargo build --verbose
-    - name: test
-      env:
-        RUST_LOG: info
-      # set PATH to ensure we use go binaries installed above
-      run: PATH=$(go env GOPATH)/bin:$PATH cargo test --verbose
-    - name: document
-      env:
-        CARGO_TARGET_DIR: "target/doc-deny"
-        RUSTDOCFLAGS: "-Dwarnings"
-      run: cargo doc
+    - name: Clippy
```

Comment:
We may want to use the rust-cache action on the lint job as well (perhaps with a separate cache key) to keep Clippy fast.

Reply:
Instead, I moved the `clippy` & other lint-like checks to a different build step to ensure they aren't poisoning the build cache (and maybe get a little perf boost from the improved parallelism). I also implemented a small change so that `clippy` can skip building the container -- it runs the build script (and I can't figure out a way to make it not do so), but it definitely doesn't need a correctly-built container image to do its job.